lib/src/room.dart

```diff
@@ -200,6 +200,7 @@ class Room {
 
   /// If something changes, this callback will be triggered. Will return the
   /// room id.
+  @Deprecated('Use `client.onSync` instead and filter for this room ID')
   final CachedStreamController<String> onUpdate = CachedStreamController();
```

Comment:
That is not quite the same, right? Since it triggers also when a member is loaded or when history is fetched?

Reply:
in this case you can listen on `client.onRoomState(roomId, EventTypes.RoomMember)`. For history fetch this is not called. But also this is called every time when something happens to a room while the value in it is always the same roomId so I would count this stream as not well thought out and would like to get rid of it

Reply:
If you call requestUser, that calls room.onUpdate. I don't think the client.onRoomState gets called in that case? While I do agree, that this doesn't seem well designed, I do think there is currently no alternative to it available?

Reply:
It does. In requestUser when `Room.onUpdate` is triggered, at the same time `Room.setState` is always called as well which then triggers `Room.client.onRoomState`

Reply:
Ah, yes, you're right. I guess this only leaves pagination, which might be fine.